feature-extensions/splunk/setup.py

```diff
@@ -0,0 +1,113 @@
+#!/usr/bin/env python2
+import subprocess
+import argparse
+import os.path
+import json
+import urllib
+
+
+class colors:
+    HEADER = '\033[95m'
+    OKBLUE = '\033[94m'
+    OKGREEN = '\033[92m'
+    WARNING = '\033[93m'
+    FAIL = '\033[91m'
+    ENDC = '\033[0m'
+    BOLD = '\033[1m'
+    UNDERLINE = '\033[4m'
+
+
+featureName = "Splunk"
```

Comment:
The idea is to have the feature installers to be completely standalone, not reaching back into the mainline installer. They will have their own Terraform scripts and their own TFVARs file, and run their own Terraform process, which you can see in the WIP Apigee example. Pretend you have no access to any files or resources outside of `feature-extensions/splunk`. 

Don't be afraid to copy things if you really need to, so we can see more clearly what needs to be shared between the main installer and the feature installers.

That does mean you need to think about how to separate out the Terraform scripts that apply only to this feature into their own folder, like with the Apigee feature installer, and pass in all input data that is needed to make that work, if feasible.

Reply:
@bleggett How the splunk terraform knows which SCM it should commit the newly created installer vars json file? It is available only in the main installer.